File: src/Entity/Certification.php
<?php

namespace App\Entity;

use App\Repository\CertificationRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=CertificationRepository::class)
 */
class Certification
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $Type;

    /**
     * @ORM\Column(type="date")
     */
    private $DatePassage;

    /**
     * @ORM\Column(type="integer")
     */
    private $Prix;

    /**
     * @ORM\Column(type="text")
     */
    private $Description;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $Nom;

  

    /**
     * @ORM\ManyToOne(targetEntity=Entreprise::class, inversedBy="certifications")
     * @ORM\JoinColumn(nullable=false)
     */
    private $IdEntreprise;

    /**
     * @ORM\OneToOne(targetEntity=DemandeCertification::class, mappedBy="IdCertif", cascade={"persist", "remove"})
     */
    private $demandeCertification;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getType(): ?string
    {
        return $this->Type;
    }

    public function setType(string $Type): self
    {
        $this->Type = $Type;

        return $this;
    }

    public function getDatePassage(): ?\DateTimeInterface
    {
        return $this->DatePassage;
    }

    public function setDatePassage(\DateTimeInterface $DatePassage): self
    {
        $this->DatePassage = $DatePassage;

        return $this;
    }

    public function getPrix(): ?int
    {
        return $this->Prix;
    }

    public function setPrix(int $Prix): self
    {
        $this->Prix = $Prix;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->Description;
    }

    public function setDescription(string $Description): self
    {
        $this->Description = $Description;

        return $this;
    }

    public function getNom(): ?string
    {
        return $this->Nom;
    }

    public function setNom(string $Nom): self
    {
        $this->Nom = $Nom;

        return $this;
    }

    public function getIdSociete(): ?int
    {
        return $this->idSociete;
    }

    public function setIdSociete(int $idSociete): self
    {
        $this->idSociete = $idSociete;

        return $this;
    }

    public function getIdEntreprise(): ?Entreprise
    {
        return $this->IdEntreprise;
    }

    public function setIdEntreprise(?Entreprise $IdEntreprise): self
    {
        $this->IdEntreprise = $IdEntreprise;

        return $this;
    }

    public function getDemandeCertification(): ?DemandeCertification
    {
        return $this->demandeCertification;
    }

    public function setDemandeCertification(DemandeCertification $demandeCertification): self
    {
        // set the owning side of the relation if necessary
        if ($demandeCertification->getIdCertif() !== $this) {
            $demandeCertification->setIdCertif($this);
        }

        $this->demandeCertification = $demandeCertification;

        return $this;
    }
}

File: src/entity/Certification.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entity;

import java.sql.Date;

/**
 *
 * @author ouaji
 */
public class Certification {
    
    private int id;
     private String type;
     private Date date_passage;
    private int  prix;
    private String description;
    private String nom;
    private int entreprise_id;

    public Certification() {
    }

    public Certification(int id, String nom) {
        this.id = id;
        this.nom = nom;
    }
    
    
    
    
    

    public Certification(String type, Date date_passage, int prix, String description, String nom, int entreprise_id) {
        this.type = type;
        this.date_passage = date_passage;
        this.prix = prix;
        this.description = description;
        this.nom = nom;
        this.entreprise_id = entreprise_id;
    }

    public Certification(int id, String type, Date date_passage, int prix, String description, String nom, int entreprise_id) {
        this.id = id;
        this.type = type;
        this.date_passage = date_passage;
        this.prix = prix;
        this.description = description;
        this.nom = nom;
        this.entreprise_id = entreprise_id;
    }

    public int getId() {
        return id;
    }

    public String getType() {
        return type;
    }

    public Date getDate_passage() {
        return date_passage;
    }

    public int getPrix() {
        return prix;
    }

    public String getDescription() {
        return description;
    }

    public String getNom() {
        return nom;
    }

    public int getEntreprise_id() {
        return entreprise_id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setDate_passage(Date date_passage) {
        this.date_passage = date_passage;
    }

    public void setPrix(int prix) {
        this.prix = prix;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setEntreprise_id(int entreprise_id) {
        this.entreprise_id = entreprise_id;
    }

  @Override
    public String toString() {
        return + id + ":" + nom;
    }
    
    
    
    
    
    
    
    
    
    
    
}
